pkg/engine/handlers/validation/validate_resource.go

```diff
@@ -250,8 +256,13 @@ func (v *validator) validateElements(ctx context.Context, foreach kyvernov1.ForE
 			return engineapi.NewRuleResponse(v.rule.Name, engineapi.Validation, msg, status), applyCount
 		}
 
+		elementCount++
 		applyCount++
 	}
+	if elementCount == 0 {
+		msg := fmt.Sprintf("validation warn: no elements that match the foreach block")
+		return engineapi.NewRuleResponse(v.rule.Name, engineapi.Validation, msg, engineapi.RuleStatusWarn), applyCount
```

Comment:
do we want to return a response or a `nil` here?

Reply:
I wasn't sure at the time, but it seems like a pr that returns a nil response has already been merged.
I missed the opportunity, didn't notice any notifications from this thread. Ah, so close to getting my first merged PR 🙂.